tests/torch/nas/test_ps_controller.py

```diff
@@ -0,0 +1,130 @@
+    config = {
+        "input_info": {"sample_size": ps_ctrl_desc.input_sizes},
+        "bootstrapNAS": {
+            "training": {
+                "batchnorm_adaptation": {
+                    "num_bn_adaptation_samples": 2
+                },
+                "elasticity": {
+                    "available_elasticity_dims": ["depth", "width"]
+                }
+            },
+        }
+    }
+    nncf_config = NNCFConfig.from_dict(config)
+    bn_adapt_args = BNAdaptationInitArgs(data_loader=create_ones_mock_dataloader(nncf_config))
+    nncf_config.register_extra_structs([bn_adapt_args])
+    return model, elasticity_ctrl, nncf_config
+
+
+def update_train_bn_adapt_section(nncf_config, bn_adapt_section_is_called):
+    if not bn_adapt_section_is_called:
+        nncf_config['bootstrapNAS']['training']['batchnorm_adaptation']['num_bn_adaptation_samples'] = 0
+
+
+# pylint: disable=protected-access
+class TestProgressiveTrainingController:
+    @pytest.mark.parametrize(('bn_adapt_section', 'is_called'), (("section_with_zero_num_samples", False),
+                                                                 ("section_with_non_zero_num_samples", True)))
+    def test_bn_adapt(self, mocker, bn_adapt_section, is_called, tmp_path, schedule_params):
+        test_desc = PSControllerTestDesc(model_creator=ThreeConvModel,
+                                     algo_params={'width': {'min_width': 1, 'width_step': 1}},
+                                     input_sizes=ThreeConvModel.INPUT_SIZE,
+                                     )
+        _, _, nncf_config = prepare_test_model(test_desc)
+        update_train_bn_adapt_section(nncf_config, is_called)
+        bn_adapt_run_patch = mocker.patch(
+            "nncf.common.initialization.batchnorm_adaptation.BatchnormAdaptationAlgorithm.run")
+        mock_model = MockModel()
+        mock_nncf_network = mocker.MagicMock(spec=NNCFNetwork)
+        mock_width_handler = mocker.MagicMock(spec=ElasticWidthHandler)
+        mock_depth_handler = mocker.MagicMock(spec=ElasticDepthHandler)
+        mock_kernel_handler = mocker.MagicMock(spec=SingleElasticityHandler)
+        handlers = OrderedDict({
+            ElasticityDim.WIDTH: mock_width_handler,
+            ElasticityDim.KERNEL: mock_kernel_handler,
+            ElasticityDim.DEPTH: mock_depth_handler,
+        })
+        mock_handler = MultiElasticityHandler(handlers, mock_nncf_network)
+        # pylint:disable=protected-access
+        mock_elasticity_ctrl = mocker.stub()
+        mock_elasticity_ctrl.multi_elasticity_handler = mock_handler
+        lr_schedule_config = {}
+        training_config = nncf_config.get('bootstrapNAS', {}).get('training', {})
+        bn_adapt_params = training_config.get('batchnorm_adaptation', {})
+        bn_adapt_algo_kwargs = get_bn_adapt_algo_kwargs(nncf_config, bn_adapt_params)
+        bn_adaptation = BatchnormAdaptationAlgorithm(**bn_adapt_algo_kwargs) if bn_adapt_algo_kwargs else None
```

Comment:
The whole point of the behavior test is lost here. 
Instead of repeating the handling `bn_adapt`, you need to verify that it's correctly handled in the training algorithm.

Propose for the training to repeat user scenario.
Just create training algorithm with mocked nncf_network and with patched run method of `BatchnormAdaptationAlgorithm`:
```python
 bn_adapt_run_patch = mocker.patch(
            "nncf.common.initialization.batchnorm_adaptation.BatchnormAdaptationAlgorithm.run")
training_algorithm = EpochBasedTrainingAlgorithm.from_checkpoint(nncf_network, bn_adapt_args,
                                                                         resuming_checkpoint_path)
```

Reply:
Thanks! Using EpochBasedTrainingAlgorithm. We also modified builder state because it wasn't saving bn adapt parameters. When we load state with "EpochBasedTrainingAlgorithm.from_checkpoint" it was using default 'num_bn_adaptation_samples' (i.e. 2000). Thanks!

Reply:
👍